### PR TITLE
Sprint Version 4.2 changes

### PR DESCRIPTION
**Pull request by babasahebk**
## Ticket
https://jira.bqsoft.com/browse/ATOM-7874

## What this does
Update API's.
1. Add Foreign Device: use `deviceSerialNumber` to send `meid`  field
2. Query Device Info: use `deviceSerialNumber` to send `meid`  field and `preferredSerialType` to indicate type of `deviceSerialNumber` as `imeiDec,meidHex,meidDec,esnDec,esnHex`
3. Add Foreign Device: use `deviceSerialNumber` to send `meid`  field and `preferredSerialType` to indicate type of `deviceSerialNumber` as `imeiDec,meidHex,meidDec,esnDec,esnHex`

### CODE REVIEW

File: lib/conduit/sprint/actions/base.rb [hunk 1]
```diff
@@ -52,6 +52,7 @@ def attributes_with_values
         # Reason for this is the layout doesnt know about these attributes so it would throw error.
         h[:xsd]         = self.class.xsd
         h[:consumer_id] = @options[:consumer_id]
+        h[:version] = @options[:version]
```
Comment (ufchrisg) [inline lib/conduit/sprint/actions/base.rb]: does this need to be skipped if `@options[:version]` is not set?

Comment (babasahebk) [reply lib/conduit/sprint/actions/base.rb]: no, this still needs to be there so that it will be available in `.erb` templates.

Comment (ufchrisg) [reply lib/conduit/sprint/actions/base.rb]: I'm just concerned about setting `h[:version]` to `nil`

Comment (babasahebk) [reply lib/conduit/sprint/actions/base.rb]: I tested it by setting it to `nil` and it works fine.

**Pull request merged**